x/reward/types/reward_pool.go

```diff
@@ -0,0 +1,49 @@
+package types
+
+import (
+	"errors"
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// NewRewardPool returns a new RewardPool object
+func NewRewardPool(
+	launchID,
+	lastRewardHeight,
+	currentRewardHeight uint64,
+	provider string,
+	coins sdk.Coins,
+) *RewardPool {
+	return &RewardPool{
+		LaunchID:            launchID,
+		LastRewardHeight:    lastRewardHeight,
+		CurrentRewardHeight: currentRewardHeight,
+		Provider:            provider,
+		Coins:               coins,
+	}
+}
+
+// Validate check the RewardPool object
+func (m RewardPool) Validate() error {
+	if m.LaunchID == 0 {
+		return errors.New("invalid launch id")
+	}
```

Comment:
```suggestion
```

I don't think we need this check since by design a launch ID can be equal to 0 if we set the first ID counter for launch ID to `0`